ecl2df/vfp.py

```diff
@@ -0,0 +1,2651 @@
+
+    # Delete rows that does not belong to any flow rate (this is
+    # possibly a by-product of not doing the stacking in an
+    # optimal way)
+    df_vfptable_stacked = df_vfptable_stacked[
+        df_vfptable_stacked["level_1"] != "DELETE"
+    ]
+
+    # Add correct column name for the flow values that we have stacked
+    cols = list(df_vfptable_stacked.columns)
+    cols[cols.index("level_1")] = "RATE"
+    df_vfptable_stacked.columns = cols
+    df_vfptable_stacked["RATE"] = df_vfptable_stacked["RATE"].astype(float)
+
+    # Sort values in correct order
+    df_vfptable_stacked.sort_values(
+        by=index_names_list + ["RATE"], ascending=True, inplace=True, ignore_index=True
+    )
+
+    return df_vfptable_stacked
+
+
+def basic_data_vfpprod2df(
+    tableno: int,
+    datum: float,
+    rate_type: VFPPROD_FLO,
+    wfr_type: WFR,
+    gfr_type: GFR,
+    alq_type: ALQ,
+    thp_type: THPTYPE,
+    unit_type: UNITTYPE,
+    tab_type: VFPPROD_TABTYPE,
+    flow_values: np.ndarray,
+    thp_values: np.ndarray,
+    wfr_values: np.ndarray,
+    gfr_values: np.ndarray,
+    alq_values: np.ndarray,
+    thp_indices: np.ndarray,
+    wfr_indices: np.ndarray,
+    gfr_indices: np.ndarray,
+    alq_indices: np.ndarray,
+    tab_data: np.ndarray,
+) -> pd.DataFrame:
+    """Return a pandas DataFrame from VFPPROD liftcurve data
+
+    Args:
+        tableno     : table number
+        datum       : datum depth
+        rate_type   : rate type used for flow values
+        wfr_type    : water fraction type
+        gfr_type    : gas fraction type
+        alq_type    : artificial lift type
+        thp_type    : thp type
+        unit_type   : unit type
+        tab_type    : type for tabulated (record) values
+        flow_values : rate values used to generate table
+        thp_values  : THP values used to generate table
+        wfr_values  : water fraction values used to generate table
+        gfr_values  : gas fraction values used to generate table
+        alq_values  : artificial lift values (if any) used to generate table
+        thp_indices : which index in thp value table a given BHP value
+                      corresponds to (1-base)
+        wfr_indices : which index in wfr value table a given BHP value
+                      corresponds to (1-base)
+        gfr_indices : which index in gfr value table a given BHP value
+                      corresponds to (1-base)
+        alq_indices : which index in alq value table a given BHP value
+                      corresponds to (1-base)
+        tab_data    : tabulated (BHP) data
+                      (ordered as thp-, wfr-, gfr-, alq- and flow-values)
+    """
+
+    # Generate list with values instead of indices
+    thp_values_list = [thp_values[i - 1] for i in thp_indices]
+    wfr_values_list = [wfr_values[i - 1] for i in wfr_indices]
+    gfr_values_list = [gfr_values[i - 1] for i in gfr_indices]
+    alq_values_list = [alq_values[i - 1] for i in alq_indices]
+
```

Comment:
Is the point of this that you need lists starting from index 1 ?

Reply:
For ecl2df this is done to replace index values (for example 1 1 1 1 at the beginning of a BHP value set) by the actual value of THP, WFR, GFR and ALQ (i.e. the value of the first element in these arrays). Do not want to store both index and value when stacking into a DataFrame.